lib/p2p/Pool.ts

```diff
@@ -49,6 +49,7 @@ interface NodeConnectionIterator {
 class Pool extends EventEmitter {
   /** The local handshake data to be sent to newly connected peers. */
   public handshakeData!: HandshakeState;
+  private pendingOutgoingConnections = new Set<string>();
```

Comment:
Maybe add a comment here to clarify what this is for like `/** A set of pub keys of nodes for which we have pending outgoing connections. */`.

Reply:
Done :speech_balloon:  :+1: